Decred Wallet/Features/Wallet Utils/SyncManager.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  SyncManager.swift
+//  Decred Wallet
+//
+// Copyright (c) 2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import Foundation
+import Dcrlibwallet
+import Signals
+
+class SyncManager: NSObject, SyncProgressListenerProtocol {
+    static let shared = SyncManager()
+    
+    var syncing = Signal<(Bool, String?)>()
+    let syncProgress =  Signal<(DcrlibwalletGeneralSyncProgress?, DcrlibwalletHeadersFetchProgressReport?)>()
+    var peers = Signal<Int32>()
+    var syncStage = Signal<(Int, String)>()
+    var connectedToNetwork: ((_ status: Bool) -> Void)?
+    
+    override init() {
+        super.init()
+        AppDelegate.walletLoader.syncer.registerSyncProgressListener(for: "\(self)", self)
+        AppDelegate.walletLoader.syncer.registerEstimatedSyncProgressListener()
+        self.connectedToNetwork?(false) // we assume wallet is offline on startup
+        self.checkNetworkConnectionForSync()
```

Comment:
Instead of beginning sync when this class is init, use a function that can be called to trigger the sync process. Currently, it is quite difficult to tell when sync starts because all it takes to start sync is to create an instance of this class somewhere, even if the instance isn't used for anything.

Reply:
resolved in 2766f8f. sync is started by calling `AppDelegate.shared.syncManager.checkNetworkConnectionForSync()` on app launch completion.